specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+            String exceptionInfo = "The script execution failed." + 
+                "\nName:" + exception.Name +
+                "\nMesssge: " + exception.Message +
+                "\nLineNumber:" + exception.LineNumber + 
+                ", ColumnNumber:" + exception.ColumnNumber;
+            Debug.WriteLine($"{exceptionInfo}");
+        }
+    }
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+/// This interface represents a JavaScript exception.
+/// If the CoreWebView2.ExecuteScriptWithResult result has IsSuccessful as false,
+/// you can use the result's Exception property to get the script exception.
+[uuid(82F22B72-1B22-403E-A0B9-A8816C9C8E45), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptException : IUnknown {
+
+  /// The line number of the source where the exception occurred.
+  /// In the JSON it is `exceptionDetail.lineNumber`.
+  /// Note that this position starts at 0.
+  [propget] HRESULT LineNumber([out, retval] UINT32* value);
+  
+  /// The column number of the source where the exception occurred.
+  /// In the JSON it is `exceptionDetail.columnNumber`.
+  /// Note that this position starts at 0.
+  [propget] HRESULT ColumnNumber([out, retval] UINT32* value);
+
+  /// The Name is the exception's class name.
+  /// In the JSON it is `exceptionDetail.exception.className`.
+  /// This is the empty string if the exception doesn't have a class name.
+  /// This can happen if the script throws a non-Error object such as `throw "abc";`
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+  /// The Message is the exception's message and potentially stack.
+  /// In the JSON it is exceptionDetail.exception.description.
+  /// This is the empty string if the exception doesn't have a description.
+  /// This can happen if the script throws a non-Error object such as throw "abc";.
+  [propget] HRESULT Message([out, retval] LPWSTR* value);
+
+  /// This will return all details of the exception as a JSON string.
+  /// In the case that script has thrown a non-Error object such as `throw "abc";`
+  /// or any other non-Error object, you can get object specific properties.
+  [propget] HRESULT ExceptionAsJSON([out, retval] LPWSTR* value);
+}
+
+/// This is the result for ExecuteScriptWithResult.
+[uuid(D2C59C5C-AD36-4CF4-87CF-2F5359F6D4CB), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptResult : IUnknown {
+
+  /// This property is true if ExecuteScriptWithResult successfully executed script with
+  /// no unhandled exceptions and the result is available in the ResultAsJson property
+  /// or via the TryGetResultAsString method.
+  /// If it is false then the script execution had an unhandled exception which you
+  /// can get via the Exception property.
+  [propget] HRESULT IsSuccess([out, retval] BOOL* value); 
+
+  /// If IsSuccess is true, then this property is the JSON representation of the result of the script execution
+  /// and otherwise returns E_INVALIDARG.
```

Comment:
E_INVALIDARG seems strange for methods that don't have any parameters.  Have we considered values like E_NOT_SET or E_NOT_FOUND to indicate empty state?  (Not sure if we're trying to restrict this to the 'common' values per https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/hresult-values , or if there is precedence for this in the API surface nearby.)

Reply:
We're open to more specific HRESULTs. Either of the suggested sound fine

Reply:
`E_UNEXPECTED` is the usual COM result for "you shouldn't have called this now, and you should have known by reading the docs."